OpenRA.Mods.RA/Activities/Teleport.cs

```diff
@@ -53,16 +55,59 @@ public override Activity Tick(Actor self)
 				if (condition.PreventsTeleport(self))
 					return NextActivity;
 
-			var bestCell = ChooseBestDestinationCell(self, destination);
-			if (bestCell == null)
-				return NextActivity;
+			if (!force)
+			{
+				var bestCell = ChooseBestDestinationCell(self, destination);
+				if (bestCell == null)
+					return NextActivity;
 
-			destination = bestCell.Value;
+				destination = bestCell.Value;
+			}
 
 			Game.Sound.Play(sound, self.CenterPosition);
 			Game.Sound.Play(sound, self.World.Map.CenterOfCell(destination));
 
 			self.Trait<IPositionable>().SetPosition(self, destination);
+			if (force)
+			{
+				if (self.Info.HasTraitInfo<HealthInfo>())
+				{
+					var health = self.Trait<Health>();
+					var givenDamage = (health.HP + health.MaxHP) / 2;
+					foreach (var a in self.World.ActorMap.GetUnitsAt(destination))
```

Comment:
The filtering here isn't sufficient. There are certain actor types (e.g. bridges, building bibs) that they should be able to return on without damage).  Use IPositionable.CanEnterCell (might be on a different interface) to check.